kclvm/error/src/diagnostic.rs

```diff
@@ -13,6 +13,7 @@ pub struct Diagnostic {
     pub level: Level,
     pub messages: Vec<Message>,
     pub code: Option<DiagnosticId>,
+    pub args: Vec<String>,
```

Comment:
1. The `args` field needs to be defined in `Message`, not in `Diagnostic`.
2. Considering the combination of lint tool error message output and error module (via `emitter`)
3. Design a generic error template instead of passing both `msg` and `args` information to emit errors